### PR TITLE
Auth urls

### PR DESCRIPTION
**Pull request by SimonLab**
ref: #6
Allow user to login with Github or Google 

image

### CODE REVIEW

File: src/Main.elm [hunk 1]
```diff
@@ -27,19 +30,21 @@ main =
 
 type Page
     = Home
-    | Auth
+    | Auth (Maybe String)
```
Comment (SimonLab) [inline src/Main.elm:33]: the `Maybe String` represent the query parameter `jwt` on the auth endpoint.
This endpoint can be `/auth` then jwt is `Nothing` or `/auth?jwt=...` then the value is `Just ...`

**Review by nelsonic — APPROVED** [review]
@SimonLab I still have a _strong_ preference for handling _all_ auth on the server including displaying the buttons in [SVG + CSS](https://github.com/dwyl/elixir-auth-google#optimised-svgcss-button) then redirecting back to the Elm App once the person has authenticated with the 3rd Party OAuth Provider. Server side auth reduces complexity and improves maintainability without sacrificing UX. 

Having reiterated that, I think the code you have written here looks _good_ and look forward to seeing your continued progress. 👍

**Comment by SimonLab** [comment]
Thanks for merging.
> I still have a strong preference for handling all auth on the server

I'll write down an issue with the requirement on https://github.com/dwyl/auth-mvp/, I think I had misunderstood where the authentication (ie google and gihtub signin button) page was supposed to be rendered.

**Comment by nelsonic** [comment]
If we render the Auth on the server it can be rendered on _any_ platform.
So if we build a Native App in Electron, Flutter or Swift etc. it's just a WebView that redirects back to the Native App. And for MVP if the person does not have a valid JWT when they attempt to make a request, the page just automatically redirects to auth.dwyl.com where the buttons are displayed in SVG+CSS. 💭 

What I'd say is _keep going_ with what you are doing because it's a _valid_ approach. 
I just want to remind you that you _might_ be overthinking it by doing the Auth interactions in Elm via API.

**Pull request merged**